group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/TargetAssignmentBuilder.java

```diff
@@ -244,16 +244,21 @@ public TargetAssignmentResult build() throws PartitionAssignorException {
         });
 
         // Prepare the topic metadata.
-        Map<Uuid, AssignmentTopicMetadata> topics = new HashMap<>();
+        Map<Uuid, PartitionMetadata> topicMetadataMap = new HashMap<>();
         subscriptionMetadata.forEach((topicName, topicMetadata) ->
-            topics.put(topicMetadata.id(), new AssignmentTopicMetadata(topicMetadata.numPartitions()))
+            topicMetadataMap.put(
+                topicMetadata.id(),
+                new PartitionMetadata(topicMetadata.partitionRacks())
```

Comment:
I am confused by the usage of PartitionMetadata only here. I think that this should be used in subscription metadata as well or not at all. It does not make sense to only use it here, right? Our goal is to avoid having to copy anything.

Reply:
Let me create a Uuid -> PartitionMetadata map in TopicMetadata itself and then pass it directly